node/chainSimulator/configs/configs.go

```diff
@@ -130,9 +111,80 @@ func CreateChainSimulatorConfigs(args ArgsChainSimulatorConfigs) (*ArgsConfigsSi
 		Configs:               configs,
 		ValidatorsPrivateKeys: privateKeys,
 		GasScheduleFilename:   gasScheduleName,
+		InitialWallets:        initialWallets,
 	}, nil
 }
 
+func generateGenesisFile(args ArgsChainSimulatorConfigs, configs *config.Configs) (*dtos.InitialWalletKeys, error) {
+	addressConverter, err := factory.NewPubkeyConverter(configs.GeneralConfig.AddressPubkeyConverter)
+	if err != nil {
+		return nil, err
+	}
+
+	initialWalletKeys := &dtos.InitialWalletKeys{
+		ShardWallets: make(map[uint32]*dtos.WalletKey),
+	}
+
+	initialAddressWithStake, err := generateWalletKeyForShard(0, args.NumOfShards, addressConverter)
+	if err != nil {
+		return nil, err
+	}
+
+	initialWalletKeys.InitialWalletWithStake = initialAddressWithStake
+
+	addresses := make([]data.InitialAccount, 0)
+	stakedValue := big.NewInt(0).Set(initialStakedEgldPerNode)
+	stakedValue = stakedValue.Mul(stakedValue, big.NewInt(int64(args.NumOfShards)+1)) // 2500 EGLD * number of nodes
+	addresses = append(addresses, data.InitialAccount{
+		Address:      initialAddressWithStake.Address,
+		StakingValue: stakedValue,
+		Supply:       stakedValue,
+	})
+
+	// generate an address for every shard
+	initialBalance := big.NewInt(0).Set(initialSupply)
+	initialBalance = initialBalance.Sub(initialBalance, stakedValue)
+
+	walletBalance := big.NewInt(0).Set(initialBalance)
+	walletBalance.Div(walletBalance, big.NewInt(int64(args.NumOfShards)))
+
+	// remainder = balance % numTotalWalletKeys
+	remainder := big.NewInt(0).Set(initialBalance)
+	remainder.Mod(remainder, big.NewInt(int64(args.NumOfShards)))
+
+	for shardID := uint32(0); shardID < args.NumOfShards; shardID++ {
+		walletKey, errG := generateWalletKeyForShard(shardID, args.NumOfShards, addressConverter)
+		if errG != nil {
+			return nil, errG
+		}
+
+		balanceForAddress := big.NewInt(0).Set(walletBalance)
+		if shardID == args.NumOfShards-1 {
```

Comment:
this can be taken out of the for loop and moved after the for finishes:
```GO
firstShardBalance := initialWalletKeys.ShardWallets[0].Balance
firstShardBalance.Add(firstShardBalance, remainder)
```

Reply:
refactored